File: OneCodeTeam/How-to-control-your-role-instances-22722/[C#]-How to control your role instances programmatically in Azure using a WCF service/C#/CSAzureRunScriptOnEachInstance/IInstanceController/IInstanceController.cs
/****************************** Module Header ******************************\
Module Name:  IInstanceController.cs
Project:      InstanceController.Interface
Copyright (c) Microsoft Corporation.
 
A WCF service contract.
 
This source is subject to the Microsoft Public License.
See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
All other rights reserved.
 
THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/
using System.Collections.Generic;
using System.ServiceModel;

namespace InstanceController.Interface
{
    [ServiceContract(Name="IInstanceControllerService",Namespace="http://www.Onecode.com")]
    public interface IInstanceController
    {
        [OperationContract]
        List<string> GetInstanceInternalEndPoint();

        [OperationContract]
        void DownLoadFileFromBlob(string ContainerName, string FileName);

        [OperationContract]
        void RunExecutableFile(string Container, string FileName);

        [OperationContract]
        bool RunScriptOnEveryInstance(string Container, string FileName);

    }
}

File: OneCodeTeam/How-to-control-your-role-instances-22722/[VB]-How to control your role instances programmatically in Azure using a WCF service/VB/VBAzureRunScriptOnEachInstance/InstanceController.Interface/IInstanceController.vb
'***************************** Module Header ******************************\
' Module Name:  IInstanceController.vb
' Project:      InstanceController.Interface
' Copyright (c) Microsoft Corporation.
' 
' A WCF service contract.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'**************************************************************************/
Imports System.ServiceModel
<ServiceContract(Name:="IInstanceControllerService", [Namespace]:="http://www.Onecode.com")> _
Public Interface IInstanceController
    <OperationContract> _
    Function GetInstanceInternalEndPoint() As List(Of String)

    <OperationContract> _
    Sub DownLoadFileFromBlob(ContainerName As String, FileName As String)

    <OperationContract> _
    Sub RunExecutableFile(Container As String, FileName As String)

    <OperationContract> _
    Function RunScriptOnEveryInstance(Container As String, FileName As String) As Boolean

End Interface